src/coreclr/src/jit/compiler.h

```diff
@@ -3761,7 +3761,8 @@ class Compiler
 
     GenTree* getArgForHWIntrinsic(var_types argType, CORINFO_CLASS_HANDLE argClass, bool expectAddr = false);
     GenTree* impNonConstFallback(NamedIntrinsic intrinsic, var_types simdType, var_types baseType);
-    GenTree* addRangeCheckIfNeeded(NamedIntrinsic intrinsic, GenTree* lastOp, bool mustExpand, int immUpperBound);
+    GenTree* addRangeCheckIfNeeded(
```

Comment:
Aren't we missing an update for `hwintrinsicxarch.cpp`?

[lookupImmUpperBound](https://github.com/dotnet/runtime/blob/master/src/coreclr/src/jit/hwintrinsicxarch.cpp#L180) is currently returning an inclusive upper bound (and I believe codegen and various other checks are dependent on that as well).

Reply:
Hmm, I though it returns upper exclusive - I will double check.